internal/common/schemafunc/string.go

```diff
@@ -0,0 +1,29 @@
+package schemafunc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+)
+
+func PlanModifyStringUpdateOnly() UpdateOnlyString {
+	return UpdateOnlyString{}
+}
+
+type UpdateOnlyString struct{}
+
+func (u UpdateOnlyString) PlanModifyString(ctx context.Context, req planmodifier.StringRequest, resp *planmodifier.StringResponse) {
```

Comment:
Prefer raising error during planning rather after plan accepted

Reply:
that's fine.  nit: maybe in a different file like plan_modifiers.go

Reply:
Thank you. Updated in af46e93db037f1858c3779895aee01a70d115dee